InfiniteScrollView.js

```diff
@@ -5,31 +5,43 @@ let ScrollableMixin = require('react-native-scrollable-mixin');
 let {
   PropTypes,
   ScrollView,
+  View,
 } = React;
 
 let cloneReferencedElement = require('react-native-clone-referenced-element');
 
 let DefaultLoadingIndicator = require('./DefaultLoadingIndicator');
 
-class InfiniteScrollView {
+class InfiniteScrollView extends React.Component {
   static propTypes = {
     ...ScrollView.propTypes,
     distanceToLoadMore: PropTypes.number.isRequired,
     canLoadMore: PropTypes.bool.isRequired,
     isLoadingMore: PropTypes.bool.isRequired,
-    onLoadMore: PropTypes.func.isRequired,
+    onLoadError: PropTypes.func,
+    onLoadMoreAsync: PropTypes.func.isRequired,
     renderLoadingIndicator: PropTypes.func.isRequired,
+    renderLoadingErrorIndicator: PropTypes.func.isRequired,
   };
 
   static defaultProps = {
-    distanceToLoadMore: 1000,
+    distanceToLoadMore: 1500,
     canLoadMore: false,
     isLoadingMore: false,
-    scrollEventThrottle: 100,
+    scrollEventThrottle: 16,
```

Comment:
I set this high because precision isn't important here and saving a little bit of battery and CPU while scrolling seemed good.